src/java/map/BinaryMask.java

```diff
@@ -168,6 +173,51 @@ public BinaryMask flipValues(float density, Symmetry symmetry) {
         return this;
     }
 
+    public BinaryMask addFloatMaskValuesWithinRangeCenteredAtLocationWithSize(FloatMask other, float minValueToConvert, float maxValueToConvert, Vector2f location, int size) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Masks not the same size: other is " + other.getSize() + " and BinaryMask is " + getSize());
+        }
+        BinaryMask maskToBeAdded = other.copy().setSize(size).convertToBinaryMask(minValueToConvert, maxValueToConvert);
+        combineWithOffset(maskToBeAdded, location, true);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public BinaryMask useBrush(Vector2f location, String brushName, float minIntensityForTrue, float maxIntensityForTrue, int size) {
+        FloatMask brush = loadBrush(brushName, random.nextLong() ,new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE));
+        addFloatMaskValuesWithinRangeCenteredAtLocationWithSize(brush, minIntensityForTrue, maxIntensityForTrue, location, size);
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public BinaryMask useBrushRepeatedlyForRandomConsecutiveExpansion(Vector2f startingLocation, String brushName, int size, int numberOfUses, float minIntensityForTrue, float maxIntensityForTrue, int maxDistanceBetweenBrushstrokeCenters) {
+        int x = (int) startingLocation.x;
+        int y = (int) startingLocation.y;
+        BinaryMask brush = loadBrush(brushName, random.nextLong(), new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE)).convertToBinaryMask(minIntensityForTrue, maxIntensityForTrue).setSize(size);
+        for (int z = 0; z < numberOfUses; z++) {
+            x += random.nextBoolean() ? random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1) : - random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1);
+            y += random.nextBoolean() ? random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1) : - random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1);
+            combineWithOffset(brush, x, y, true);
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public BinaryMask connectSpawnsWithRandomConsecutiveBrushUse(ArrayList<Spawn> spawns, int percentChanceToAttemptConnectionPerOddNumberedSpawn, String brushName, int size, int numberOfUsesBatchSize, float minIntensityForTrue, float maxIntensityForTrue, int maxDistanceBetweenBrushstrokeCenters) {
+        for (int z = 0; z < spawns.size() / 2; z+=2) {
+            if(percentChanceToAttemptConnectionPerOddNumberedSpawn > random.nextInt(100) - 1) {
```

Comment:
percentChance should be a float and then use random.nextFloat for condition check

Reply:
changed